internal/commands/pull/command.go

```diff
@@ -60,32 +61,58 @@ func (cmd *Command) Handler(profile *user.Profile, ui terminal.UI, clients cli.C
 		return err
 	}
 
-	pathTarget, zipPkg, err := cmd.doExport(profile, clients.Realm, app.GroupID, app.ID)
+	clientZipPkgs, err := cmd.inputs.resolveClientTemplates(ui, clients.Realm, app.GroupID, app.ID)
 	if err != nil {
 		return err
 	}
 
-	pathRelative, err := filepath.Rel(profile.WorkingDirectory, pathTarget)
+	projectPath, zipPkg, err := cmd.doExport(profile, clients.Realm, app.GroupID, app.ID)
 	if err != nil {
 		return err
 	}
 
-	proceed, err := checkAppDestination(ui, pathTarget)
+	var pathFrontend string
+	pathBackend := projectPath
+	if len(clientZipPkgs) != 0 {
+		pathFrontend = filepath.Join(projectPath, local.FrontendPath)
+		if proceed, err := checkPathDestination(ui, pathFrontend); err != nil {
+			return err
+		} else if !proceed {
+			return nil
+		}
+		pathBackend = filepath.Join(projectPath, local.BackendPath)
+	}
+
+	// App path
+	proceed, err := checkPathDestination(ui, pathBackend)
 	if err != nil {
 		return err
 	} else if !proceed {
 		return nil
 	}
 
+	pathRelative, err := filepath.Rel(profile.WorkingDirectory, projectPath)
+	if err != nil {
+		return err
+	}
+
 	if cmd.inputs.DryRun {
-		ui.Print(
-			terminal.NewTextLog("No changes were written to your file system"),
-			terminal.NewDebugLog("Contents would have been written to: %s", pathRelative),
-		)
+		if len(cmd.inputs.TemplateID) != 0 {
+			ui.Print(
+				terminal.NewTextLog("No changes were written to your file system"),
```

Comment:
[DRY] we can build the list of logs to print (to ensure a consistent first message)

```go
logs := make([]terminal.Log, 0, 3)
logs = append(logs, terminal.NewTextLog("No changes were written to your file system"))

if len(clientZipPkgs) != 0 { // NOTE: I think its important we stick with the same check throughout here
  logs = append(logs,
    terminal.NewDebugLog("App contents would have been written to: %s", filepath.Join(pathRelative, local.BackendPath)),
    terminal.NewDebugLog("Template contents would have been written to: %s", filepath.Join(pathRelative, local.FrontendPath)),
  )
} else {
  logs = append(logs, terminal.NewDebugLog("Contents would have been written to: %s", pathRelative))
}
ui.Print(logs...)
return nil
```